server/src/test/java/org/elasticsearch/monitor/fs/FsHealthServiceTests.java

```diff
@@ -231,6 +234,132 @@ public void testFailsHealthOnSinglePathWriteFailure() throws IOException {
         }
     }
 
+    public void testFailsHealthOnMissingLockFile() throws IOException {
```

Comment:
Thorough tests 😄 However they're not really testing anything in the `FsHealthService` so much as testing the details of the implementation of the `NativeFSLock`. Let's just have one of these here, and maybe consider filling in any gaps in Lucene's [TestNativeFSLockFactory](https://github.com/apache/lucene-solr/blob/master/lucene/core/src/test/org/apache/lucene/store/TestNativeFSLockFactory.java) separately.

Reply:
Thanks, i am thinking to keep two of them where one throws and `IOException` and another for `AlreadyClosedException`.

Reply:
No, one is all we need here.

`NodeEnvironmentTests` would be the right place to verify that `NodeEnvironment#assertEnvIsLocked` throws an `IllegalStateException` in both of those cases. I think we don't do that today, but again that's a question for a separate PR.

Reply:
ok, Got it.